### PR TITLE
[RFR] Fix entity list retrieval

### PR DESCRIPTION
**Pull request by jpetitcolas**
Tests will come in another PR to not be blocking further.

### CODE REVIEW

File: app/Services/RestWrapper.js [hunk 1]
```diff
@@ -36,7 +36,8 @@ class RestWrapper {
             .then((response) => {
                 return {
                     data: response().data,
-                    totalCount: response.headers()['x-total-count']
+                    totalCount: response().data.totalCount,
```
Comment (jeromemacias) [inline app/Services/RestWrapper.js]: Why do you ignore the header ?
You should check header first, then `totalCount`, then do a count of data elements.

File: app/View/List.js [hunk 2]
```diff
@@ -60,14 +60,11 @@ class ListView extends React.Component {
...
-    onLoadFailure(response) {
-        let body = response.data;
-        if ('object' === typeof message) {
-            body = JSON.stringify(body);
-        }
-
-        Notification.log(`Oops, an error occured during data fetching  : (code: ${response.status}) ${body}`,
-            {addnCls: 'humane-flatty-error'});
+    onLoadFailure(error) {
+        console.error(error);
```
Comment (jeromemacias) [inline app/View/List.js]: You must check if `console` is available.

File: app/View/List.js [hunk 3]
```diff
@@ -60,14 +60,11 @@ class ListView extends React.Component {
...
+        Notification.log(`Error while fecthing data: ${error.message}.`, {
```
Comment (jeromemacias) [inline app/View/List.js]: `error` is sent by `Restful.js` ?
What if it's a 404, not a 500, or other error types ?

File: app/View/List.js (hunk 2 above)
Comment (jpetitcolas) [inline app/View/List.js]: In which case may it be unavailable?

Comment (jeromemacias) [inline app/View/List.js]: Old browsers...

File: app/View/List.js (hunk 3 above)
Comment (jpetitcolas) [inline app/View/List.js]: I got a JS error here, not a restful one.

Comment (jeromemacias) [inline app/View/List.js]: If you misconfigure your entity name, you will have a 404 error by Restful, same for 500.

File: app/Services/RestWrapper.js (hunk 1 above)
Comment (jpetitcolas) [inline app/Services/RestWrapper.js]: All these checks are done in `admin-config` project, in the [ReadQueries](https://github.com/marmelab/admin-config/blob/master/lib/Queries/ReadQueries.js#L49). We should provide it the same structure in both ng-admin and react-admin.

Comment (jeromemacias) [inline app/Services/RestWrapper.js]: Ok, did'nt notice it, good catch!

**Pull request merged**